delfin/task_manager/manager.py

```diff
@@ -39,6 +49,49 @@ def __init__(self, service_name=None, *args, **kwargs):
         self.alert_sync = alerts.AlertSyncTask()
         super(TaskManager, self).__init__(*args, **kwargs)
 
+    def periodic_performance_collect(self):
+        LOG.info("Scheduled performance collection starting.")
+
+        # Load the scheduler configuration file
+        try:
+            with open(CONF.scheduler.config_path) as f:
+                data = json.load(f)
+        except FileNotFoundError as e:
+            raise exception.ConfigNotFound(e)
+
+        # create the object of periodic scheduler
+        schedule = BlockingScheduler()
+
+        # create the objet to StorageController class, so that
+        # methods of that class can be called by scheduler
+        storage_cls = StorageController()
+
+        # parse the scheduler configuration file and start the task
+        # for each storage
+        for storage in data.get("storages"):
+            storage_id = storage.get('id')
+            for resource in storage.keys():
+
+                if resource == 'id':
+                    continue
+
+                resource_type = storage.get(resource)
+                if (resource_type.get('perf_collection') and
+                        resource_type.get('interval') > MIN_INTERVAL):
```

Comment:
Couple of questions:
a) what happens when the resource_type.get('interval') < MIN_INTERVAL, job won't be added but scheduler will start, is that fine?
b) this 'interval' is device managed right?

Reply:
Yes, job won't be added to scheduler so it will never execute.
interval, is the time to collect metrics periodically.

Reply:
Ok